### ISSUE TITLE
Fix alt text enhancements

### ISSUE THREAD

**Issue by lukefretwell** [proudcity/wp-proudcity#2550]
## Builds

- [staging](https://console.cloud.google.com/cloud-build/builds;region=global/8ca0aa9f-84f8-4409-a302-edb4e4cbca04?project=proudcity-1184)
- [prod](https://console.cloud.google.com/cloud-build/builds;region=global/5f43bfc0-e6e3-4cb9-9d36-c996af18833a?project=proudcity-1184)

## Issue

**Source**
The source of the issue (ex: Customer: #CustomerName#).

ProudCity

**Is your feature request related to a problem? Please describe.**

Only to make certain items invisible to admins.

**What feature would you like to improve**
A clear and concise description of the current feature.

Hide:

- [x] Current settings box on dashboard page
- [x] Settings tab
- [x] Notices? (example below references settings)
- [x] Warnings? (example below references settings)

**Describe the solution you'd like**

**Additional context**

Current settings:

![Screenshot 2024-05-14 at 10-45-41 Fix Alt Text ‹ ProudCity — WordPress](https://github.com/proudcity/wp-proudcity/assets/1569463/a99392c4-6184-4f39-9891-83f9f2b1ffb4)

Settings tab:

![Screenshot 2024-05-14 at 10-46-07 Settings](https://github.com/proudcity/wp-proudcity/assets/1569463/cac6d8fa-6af7-4a25-81ec-b6ef005a12d1)

Notices:

![Screenshot 2024-05-14 at 10-45-56 Alt Text](https://github.com/proudcity/wp-proudcity/assets/1569463/d33805da-9b36-4c77-a0fb-03adeba8bc17)

Warning:

![Screenshot 2024-05-14 at 11-58-07 Fix Alt Text ‹ ProudCity — WordPress](https://github.com/proudcity/wp-proudcity/assets/1569463/34662e01-72e8-42f8-a585-9e3fea68144c)

## Deploy

- [ ] add new settings to allow editors
```php
{
  "option": "fix_alt_text",
  "db_version": "1.7.0",
  "db_version_history": [],
  "access_tool_roles": [
    "editor"
  ],
  "access_settings_roles": [
    "editor"
  ],
  "debug": false,
  "blocks": [
    "core/image",
    "core/media-text",
    "core/gallery"
  ],
  "others": [
    "Media Library"
  ],
  "scan_post_types": [
    "attachment",
    "page",
    "wp_block",
    "post",
    "wp_template_part",
    "wp_template"
  ],
  "scan_taxonomies": [
    "category",
    "post_tag",
    "wp_template_part_area",
    "wp_theme"
  ],
  "scan_users": true,
  "site_id": 1
}
```

**Comment by kevindherman** [comment]
When ready:

- [ ]give WP editors permissions
- Link to better documentation

**Comment by curtismchale** [comment]
@kevindherman @lukefretwell this is up in Beta for testing. @kevindherman what specifically did you mean when you asked for a link to better documentation? I'm not sure where I would put that.

**Comment by kevindherman** [comment]
@curtismchale awesome. I'm trying to get the first scan going but it's just spinning. Are you able to generate a scan?

For documentation, I think the idea was to add links to support articles. There will likely be a future article for the specific tool but for now, this is the newest and best alt text article: https://help.proudcity.com/alt-text-and-accessible-images/

@lukefretwell I think you had more specific ideas on helpful doc placement? 

Maybe it's phase 1 using the existing article to get this out and then we add more useful documentation down the line.

**Comment by kevindherman** [comment]
@curtismchale Beta has a lot of images and my guess is that many don't have alt text so I wonder if the size of the scan is part of the issue. Scanning San Rafael and Petaluma respectively took a couple of goes.

**Comment by curtismchale** [comment]
@kevindherman @lukefretwell so this still isn't quite running on Beta, but it doe work on [another test site](https://general-plan-santa-ana-ca.proudcity.com/wp-admin/tools.php?page=fix-alt-text&tab=dashboard). I assume that @kevindherman idea above is correct, there are lots of images on `beta` and it may just be bogging things down. I'm letting the Fix Alt Text plugin run on Beta for a while to see if it works eventually.

For now check out [the other test site](https://general-plan-santa-ana-ca.proudcity.com/wp-admin/tools.php?page=fix-alt-text&tab=dashboard) to verify that this works as expected.

**Comment by kevindherman** [comment]
@curtismchale @lukefretwell this is great.

**Dashboard:**
@lukefretwell there are extra items in the dashboard report which makes the total issues found number bigger. If you click on "alt text missing" it brings you to the filtered list.  We talked about limiting that but I think it's helpful as long as we provide context. 

@curtismchale are we able to limit to just alt text missing? That could be a good phase 1 and then the other items we could role out later if @lukefretwell is concerned about freaking people out with a big number of issues when alt text missing is the main problem.

**Total images**
@curtismchale I think it's worth testing the ballpark of when a site has too many images for the tool. Can do a big bulk delete of images on BETA to bring it down say 250-500 images and see where the tool starts scanning right. @lukefretwell this could be a good way of getting customers to clean out/delete old images. "If you have more than 2000 image files, the scan can time out. Accumulating 2000 images over the course of years is also weighing down your site, consider deleting images from your media library to bring you under #### to have the scan work better and optimize your site."

**Comment by lukefretwell** [comment]
I'm not as concerned about the number of images causing alarm. Would be great to understand (and see) what we can hide per specs here: https://github.com/proudcity/wp-proudcity/issues/2550#issue-2296219459

**Comment by kevindherman** [comment]
@lukefretwell it looks like everything was edited out from your comments. Do you think this message/alert should also be edited out? 
![Screen Shot 2024-06-20 at 2 28 23 PM](https://github.com/proudcity/wp-proudcity/assets/6032426/e0f139ec-0b37-4487-9ca5-a4cc34529d5a)

**Comment by lukefretwell** [comment]
@kevindherman yes

**Comment by curtismchale** [comment]
@kevindherman I'm not sure that the amount of images is the problem with Beta. I just ran it on my site with 3500 images and it took about 1 minute to run. It showed progress within about 10 seconds. I'm deploying to proudcity now to test there again to see if the image processing works.

I don't think we should limit the report either. It's possible with CSS to hid things, but it's just a list of items so I have to use CSS that counts the items. If they ever change the order or add one item the CSS isn't going to work as we expect. I think we should focus on education, good alt text is a requirement they have to meet. I don't think hiding the issue serves customers well at all.

### Ballpark image numbers

We know that 3500 work on my site. I just ran it locally on a client site with 4500 images and it worked fine. I'm grabbing a copy of a client site with 11k images to test as well. That's taking a bit so I'll update if it doesn't work.

## What we hide

@lukefretwell we hide everything you asked for except the bell icon because I missed that arrow. I'll do that now. We only hide that if you're **not** a WordPress Administrator. I know there is 1 or 2 customers that have this access but for the vast majority, everything you wanted is hidden.

I'll do a new build shortly to hide the bell as well.

**Comment by kevindherman** [comment]
I'm totally fine with not limited the report, it was just discussed previously. 

Sounds like we could be close to getting out.

**Comment by curtismchale** [comment]
Deployed

Composer: https://github.com/proudcity/wp-proudcity/releases/tag/2024.06.19.0955
Builds: https://github.com/proudcity/proud-recipes/releases/tag/2024.06.19.0955
